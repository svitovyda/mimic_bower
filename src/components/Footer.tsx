import * as React from "react";

export const Footer: React.FC = () => {
  return <div></div>;
};
