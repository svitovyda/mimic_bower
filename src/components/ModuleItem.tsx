import type { Package } from "../models/Package";
import * as React from "react";

export interface ModuleItemProp {
  packageElement: Package;
}

export const ModuleItem: React.FC<ModuleItemProp> = ({ packageElement }: ModuleItemProp) => {
  return (
    <div>
      {packageElement.name} {packageElement.owner} {packageElement.stars}
    </div>
  );
};
