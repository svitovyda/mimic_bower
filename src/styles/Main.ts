import styled from 'styled-components';

export const MainContainer = styled.div`
  max-width: 1000px;
  margin: 0 auto;
  padding: 0 10px;
  display: flex;
  @media only screen and (max-width: 680px) {
    flex-direction: column;
  }
`;

export const RightContainer = styled.div`
  width: 100%;

  @media only screen and (min-width: 681px) {
    flex-grow: 1;
  }
`;
