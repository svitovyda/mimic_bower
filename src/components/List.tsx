import type { Package } from '../models/Package';
import { ModuleItem } from './ModuleItem';
import * as React from 'react';

export interface ListProps {
  data: Package[];
}

export const List: React.FC<ListProps> = ({ data }: ListProps) => {
  return (
    <div>
      {data.map((p, i) => (
        <ModuleItem key={`${i}-${p.name}-${p.platform}`} packageElement={p} />
      ))}
    </div>
  );
};
