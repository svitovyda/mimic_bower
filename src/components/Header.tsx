import { HeaderContainer, HeaderLogo, HeaderSubtitle, HeaderTitle } from '../styles/Header';
import * as React from 'react';

export const Header: React.FC = () => {
  return (
    <header>
      <HeaderContainer>
        <HeaderLogo />
        <HeaderTitle>Bower Search</HeaderTitle>
        <HeaderSubtitle>
          Powered by{' '}
          <a href="https://libraries.io/" target="_blank" rel="noreferrer">
            libraries.io
          </a>
        </HeaderSubtitle>
      </HeaderContainer>
    </header>
  );
};
