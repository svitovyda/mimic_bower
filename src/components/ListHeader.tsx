/* eslint-disable jsx-a11y/anchor-is-valid */
import type { SortType } from '../services/ApiServise';
import { ListHeaderContainer, NameLink, OwnerLink, StarsLink } from '../styles/ListHeader';
import * as React from 'react';

export interface ListHeaderProps {
  onSort: (sortBy?: SortType) => void;
}

export const ListHeader: React.FC<ListHeaderProps> = React.memo(({ onSort }: ListHeaderProps) => {
  return (
    <ListHeaderContainer>
      <NameLink onClick={() => onSort()}>Name</NameLink>
      <OwnerLink>Owner</OwnerLink>
      <StarsLink onClick={() => onSort('stars')}>Stars</StarsLink>
    </ListHeaderContainer>
  );
});

ListHeader.displayName = 'ListHeader';
