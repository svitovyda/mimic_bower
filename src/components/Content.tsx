import { useContentFetcher } from "../hooks/contentFetcher";
import { usePagination } from "../hooks/usePagination";
import type { Package } from "../models/Package";
import { List } from "./List";
import { Pagination } from "./Pagination";
import { LoaderAnimation } from "./ui/LoaderAnimation";
import configJson from "config";
import * as React from "react";

const ItemsPerPage = configJson.searchItemsPageMaxSize;

export interface ContentProps {
  query: string;
}

export const Content: React.FC<ContentProps> = ({ query }: ContentProps) => {
  const [fetchState, setQuery] = useContentFetcher("");
  const [paginationState, navigateToPage] = usePagination(fetchState.data);
  const [pageData, setPageData] = React.useState<Package[]>([]);

  React.useEffect(() => {
    setQuery(query);
    setPageData([]);
  }, [query, setQuery]);

  React.useEffect(() => {
    if (fetchState.data.length === 0) setPageData([]);
    else {
      const pageStart = (paginationState.currentPage - 1) * ItemsPerPage;

      setPageData(fetchState.data.slice(pageStart, pageStart + ItemsPerPage));
    }
  }, [fetchState.data, paginationState]);

  if (fetchState.isLoading) return <LoaderAnimation />;
  if (fetchState.isError) return <div>Sorry, there was an error while fetching repositories</div>;
  return (
    <div>
      <List data={pageData} />
      <Pagination
        currentPage={paginationState.currentPage}
        totalPages={paginationState.totalPages}
        navigateToPage={navigateToPage}
      />
    </div>
  );
};
