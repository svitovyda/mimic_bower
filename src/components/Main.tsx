import { MainContainer, RightContainer } from "../styles/Main";
import { Content } from "./Content";
import { Footer } from "./Footer";
import { Search } from "./Search";
import { Sidebar } from "./Sidebar";
import * as React from "react";

export const Main: React.FC = () => {
  const [query, setQuery] = React.useState<string>("");

  const onNewQuery = React.useCallback(
    (newQuery: string) => {
      setQuery(newQuery);
    },
    [setQuery]
  );

  return (
    <MainContainer>
      <Sidebar />
      <RightContainer>
        <Search onQueryChanged={onNewQuery} />
        <Content query={query} />
        <Footer />
      </RightContainer>
    </MainContainer>
  );
};
