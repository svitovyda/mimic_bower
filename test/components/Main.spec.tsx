import { Main } from '../../src/components/Main';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import * as React from 'react';

describe('Main', () => {
  it('renders without error', () => {
    const component = render(<Main />);

    expect(component).toMatchSnapshot();
  });
});
