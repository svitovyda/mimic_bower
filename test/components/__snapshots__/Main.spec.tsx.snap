// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders without error 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  max-width: 1000px;
  margin: 0 auto;
  padding: 0 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  width: 100%;
}

.c2 {
  cursor: pointer;
  outline: 0;
  display: inline-block;
  font-weight: bold;
  line-height: 1.5;
  height: 45px;
  text-align: center;
  background-color: white;
  border: 1px solid #ccc;
  border-radius: 2px;
  padding: 6px 12px;
  font-size: 18px;
  color: #00acee;
}

.c2:hover:enabled {
  border-color: #ef5734;
}

.c2:disabled {
  color: lightgray;
  border-color: lightgray;
}

.c6 {
  display: block;
}

.c7 {
  display: inline-block;
  padding-block: 1px;
  padding-inline: 2px;
  padding: 12px 10px;
  cursor: text;
  height: 45px;
  border: 0;
  border-radius: 1px;
  box-shadow: 0 0 1px #ccc;
  font-size: 18px;
  font-weight: 400;
  width: 100%;
  margin-bottom: 10px;
  outline-width: 0;
  color: inherit;
}

.c3 {
  width: 250px;
  margin: 0px auto 20px auto;
  display: none;
}

.c4 {
  width: 100%;
  display: none;
}

@media only screen and (max-width:680px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media only screen and (min-width:681px) {
  .c5 {
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
  }
}

@media only screen and (max-width:680px) {
  .c3 {
    display: inline-block;
  }
}

@media only screen and (max-width:680px) {
  .c1 {
    text-align: center;
    width: 100%;
  }
}

@media only screen and (min-width:681px) {
  .c1 {
    width: 200px;
    -webkit-flex-shrink: 0;
    -ms-flex-negative: 0;
    flex-shrink: 0;
  }
}

@media only screen and (min-width:681px) {
  .c4 {
    display: block;
  }
}

<body>
    <div>
      <div
        class="c0"
      >
        <div
          class="c1"
        >
          <button
            class="c2 c3"
          >
            Menu
          </button>
          <div
            class="c4"
          >
            <ul>
              <li>
                <a
                  href="#"
                >
                  Home
                </a>
              </li>
              <li>
                <a
                  href="#"
                >
                  Creating Packages
                </a>
              </li>
              <li>
                <a
                  href="#"
                >
                  API
                </a>
              </li>
              <li>
                <a
                  href="#"
                >
                  Configuration
                </a>
              </li>
              <li>
                <a
                  href="#"
                >
                  Pluggable Resolvers
                </a>
              </li>
              <li>
                <a
                  href="#"
                >
                  Tools
                </a>
              </li>
              <li>
                <a
                  href="#"
                >
                  About
                </a>
              </li>
            </ul>
          </div>
        </div>
        <div
          class="c5"
        >
          <div
            class="c6"
          >
            <input
              autocomplete="off"
              class="c7"
              maxlength="100"
              placeholder="Search..."
              type="search"
              value=""
            />
          </div>
          <div>
            <div />
            <div>
              <button
                class="c2"
                disabled=""
              >
                ← Previous
              </button>
              <button
                class="c2"
                disabled=""
              >
                Next →
              </button>
            </div>
          </div>
          <footer>
            <p>
              <a
                href="https://github.com/bower/bower.github.io/issues"
                rel="noreferrer"
                target="_blank"
              >
                Help improve these docs. Open an issue or pull request.
              </a>
            </p>
            <p>
              Our supporters: TBA
            </p>
          </footer>
        </div>
      </div>
    </div>
  </body>,
  "container": .c0 {
  max-width: 1000px;
  margin: 0 auto;
  padding: 0 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  width: 100%;
}

.c2 {
  cursor: pointer;
  outline: 0;
  display: inline-block;
  font-weight: bold;
  line-height: 1.5;
  height: 45px;
  text-align: center;
  background-color: white;
  border: 1px solid #ccc;
  border-radius: 2px;
  padding: 6px 12px;
  font-size: 18px;
  color: #00acee;
}

.c2:hover:enabled {
  border-color: #ef5734;
}

.c2:disabled {
  color: lightgray;
  border-color: lightgray;
}

.c6 {
  display: block;
}

.c7 {
  display: inline-block;
  padding-block: 1px;
  padding-inline: 2px;
  padding: 12px 10px;
  cursor: text;
  height: 45px;
  border: 0;
  border-radius: 1px;
  box-shadow: 0 0 1px #ccc;
  font-size: 18px;
  font-weight: 400;
  width: 100%;
  margin-bottom: 10px;
  outline-width: 0;
  color: inherit;
}

.c3 {
  width: 250px;
  margin: 0px auto 20px auto;
  display: none;
}

.c4 {
  width: 100%;
  display: none;
}

@media only screen and (max-width:680px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media only screen and (min-width:681px) {
  .c5 {
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
  }
}

@media only screen and (max-width:680px) {
  .c3 {
    display: inline-block;
  }
}

@media only screen and (max-width:680px) {
  .c1 {
    text-align: center;
    width: 100%;
  }
}

@media only screen and (min-width:681px) {
  .c1 {
    width: 200px;
    -webkit-flex-shrink: 0;
    -ms-flex-negative: 0;
    flex-shrink: 0;
  }
}

@media only screen and (min-width:681px) {
  .c4 {
    display: block;
  }
}

<div>
    <div
      class="c0"
    >
      <div
        class="c1"
      >
        <button
          class="c2 c3"
        >
          Menu
        </button>
        <div
          class="c4"
        >
          <ul>
            <li>
              <a
                href="#"
              >
                Home
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Creating Packages
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                API
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Configuration
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Pluggable Resolvers
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Tools
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                About
              </a>
            </li>
          </ul>
        </div>
      </div>
      <div
        class="c5"
      >
        <div
          class="c6"
        >
          <input
            autocomplete="off"
            class="c7"
            maxlength="100"
            placeholder="Search..."
            type="search"
            value=""
          />
        </div>
        <div>
          <div />
          <div>
            <button
              class="c2"
              disabled=""
            >
              ← Previous
            </button>
            <button
              class="c2"
              disabled=""
            >
              Next →
            </button>
          </div>
        </div>
        <footer>
          <p>
            <a
              href="https://github.com/bower/bower.github.io/issues"
              rel="noreferrer"
              target="_blank"
            >
              Help improve these docs. Open an issue or pull request.
            </a>
          </p>
          <p>
            Our supporters: TBA
          </p>
        </footer>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
